pkg/extpromql/parser.go

```diff
@@ -43,16 +43,7 @@ func ParseMetricSelector(input string) ([]*labels.Matcher, error) {
 		return nil, fmt.Errorf("expected type *parser.VectorSelector, got %T", expr)
 	}
 
-	matchers := make([]*labels.Matcher, len(vs.LabelMatchers))
-	for i, lm := range vs.LabelMatchers {
-		matchers[i] = &labels.Matcher{
-			Type:  lm.Type,
-			Name:  lm.Name,
-			Value: lm.Value,
-		}
-	}
-
-	return matchers, nil
+	return vs.LabelMatchers, nil
```

Comment:
I think there is already a function in Prometheus for this: https://github.com/prometheus/prometheus/blob/789c9b1a5e455850ed9b3c89cafb37df75ce1e50/promql/parser/ast.go#L333